src/simple-dom/document/node.ts

```diff
@@ -34,12 +34,19 @@ export default class Node {
   public firstChild: Node | null = null;
   public lastChild: Node | null = null;
 
-  public childNodes: {
-    item(index: number): Node | null;
-  };
+  private _childNodes: ChildNodes | undefined = undefined;
 
   constructor(public nodeType: NodeType, public nodeName: string, public nodeValue: string | null) {
-    this.childNodes = new ChildNodes(this);
+  }
+
+  public get childNodes(): {
```

Comment:
Is there a reason we can't use `ChildNodes` are the return type annotation here?

Reply:
Yes, I don't want to leak the constructor

Reply:
You mean because using it in the type position means that you have to export the concrete implementation as well? Might be nice to have a feature in TypeScript that allows you to export just the type definition in cases like this where a single construct produces both.